tests/slsa_analyzer/checks/test_build_service_check.py

```diff
@@ -116,6 +132,48 @@ def test_build_service_check(self) -> None:
         gradle_build_ci.dynamic_data["ci_services"] = [ci_info]
         assert check.run_check(gradle_build_ci, check_result) == CheckResultType.PASSED
 
+        # Use poetry in CI to build the artifact.
+        poetry_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
+        poetry_build_ci.dynamic_data["build_spec"]["tool"] = poetry
+        bash_commands["commands"] = [["poetry", "build"]]
+        poetry_build_ci.dynamic_data["ci_services"] = [ci_info]
+        assert check.run_check(poetry_build_ci, check_result) == CheckResultType.PASSED
+
+        # Use pip in CI to build the artifact.
+        pip_module_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
+        pip_module_build_ci.dynamic_data["build_spec"]["tool"] = pip
+        bash_commands["commands"] = [["pip", "install"]]
+        pip_module_build_ci.dynamic_data["ci_services"] = [ci_info]
+        assert check.run_check(pip_module_build_ci, check_result) == CheckResultType.PASSED
+
+        # Use flit in CI to build the artifact.
+        flit_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
+        flit_build_ci.dynamic_data["build_spec"]["tool"] = pip
+        bash_commands["commands"] = [["flit", "build"]]
+        flit_build_ci.dynamic_data["ci_services"] = [ci_info]
+        assert check.run_check(flit_build_ci, check_result) == CheckResultType.PASSED
+
+        # Use pip as a module in CI to build the artifact.
+        pip_module_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
+        pip_module_build_ci.dynamic_data["build_spec"]["tool"] = pip
+        bash_commands["commands"] = [["python", "-m", "pip", "install"]]
+        pip_module_build_ci.dynamic_data["ci_services"] = [ci_info]
+        assert check.run_check(pip_module_build_ci, check_result) == CheckResultType.PASSED
+
+        # Use pip as a module incorrectly in CI to build the artifact.
+        pip_module_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
+        pip_module_build_ci.dynamic_data["build_spec"]["tool"] = pip
+        bash_commands["commands"] = [["python", "pip", "install"]]
+        pip_module_build_ci.dynamic_data["ci_services"] = [ci_info]
+        assert check.run_check(pip_module_build_ci, check_result) == CheckResultType.FAILED
+
+        # Use pip as a module in CI with invalid goal to build the artifact.
+        pip_module_build_ci = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
```

Comment:
When you adjust the attribute names in the spec, please update the variable names here too.

Reply:
Done in [9c167f2](https://github.com/oracle-samples/macaron/pull/67/commits/9c167f271c3b880c3db95d6de5fc7bb51bdfcf3d).